### PR TITLE
Add service/timer for checking restart

### PR DESCRIPTION
**Pull request by neha170**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 
Add service/timer for checking restart

###### Change Log  
- Adds a service that checks if a system restart is required due to recent package updates. If restart is required, it touches /var/run/reboot-required file (default behavior in ubuntu)
- Corresponding timer runs everyday at 08:00 UTC

###### Does this affect the toolchain?  
NO

###### Test Methodology
- Local build. Tested functionality on Mariner marketplace gen1 image on hyper-v.

### CODE REVIEW

File: SPECS/check-restart/check-restart.spec [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+%systemd_preun check-restart.{service,timer}
+
+%files
+%{_unitdir}/%{name}.service
+%{_unitdir}/%{name}.timer
+%{_bindir}/%{name}.sh
+
+%changelog
+* Mon Aug 02 2021 Neha Agarwal <nehaagarwal@microsoft.com> - 1.0.0-1
+- Initial version of the check-restart package
```
Comment (PawelWMS) [inline SPECS/check-restart/check-restart.spec:44]: If it's our own spec, please use the phrase `Original version for CBL-Mariner.` - that's what we're trying to follow in other specs.

File: SPECS/check-restart/check-restart.spec [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Name:           check-restart
+Version:        1.0.0
+Release:        1%{?dist}
+Summary:        Create new systemd timer to check for system restart 
+License:        MIT
+Source0:        check-restart.service
+Source1:        check-restart.sh
+Source2:        check-restart.timer
+Vendor:         Microsoft Corporation
```
Comment (PawelWMS) [inline SPECS/check-restart/check-restart.spec:11]: Please run the linter to put the tags in the correct order.

File: SPECS/check-restart/check-restart.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+OUTPUT=$(needs-restarting -r)
```
Comment (PawelWMS) [inline SPECS/check-restart/check-restart.sh:1]: Curious: which package provides the `needs-restarting` command? We should add it to the `Requires` for the new package.

**Review by PawelWMS — CHANGES_REQUESTED** [review]

File: SPECS/check-restart/check-restart.sh (hunk 3 above)
Comment (neha170) [reply SPECS/check-restart/check-restart.sh:1]: It needs dnf-utils. Added to Requires section.

**Review by PawelWMS — APPROVED** [review]

File: SPECS/core-packages/core-packages.spec [hunk 4]
```diff
@@ -83,6 +80,10 @@ Requires:       vim
 Requires:       wget
 Requires:       xz
 Requires:       zlib
+%ifarch x86_64
+# Temporarily exclude aarch64 from including the microsoft repo until content is available in the repo
+Requires:       mariner-repos-microsoft
+%endif
```
Comment (hbeberman) [inline SPECS/core-packages/core-packages.spec:86]: You should move this section back up to where it was right below `Requires: mariner-repos`, it makes more sense there alphabetically, the linter just gets confused by the %ifarch.

**Review by hbeberman — APPROVED** [review]
LGTM. Just one nitpick about overzealous linting.

File: SPECS/core-packages/core-packages.spec (hunk 4 above)
Comment (neha170) [reply SPECS/core-packages/core-packages.spec]: Done

**Pull request merged**